src/main/java/ru/aasmc/financialmetrics/metrics/LatestSumCalculator.kt

```diff
@@ -0,0 +1,30 @@
+package ru.aasmc.financialmetrics.metrics
+
+import ru.aasmc.financialmetrics.filters.Filter
+import ru.aasmc.financialmetrics.model.Instrument
+import java.util.*
+
+class LatestSumCalculator(
+    val name: String,
+    override val filters: List<Filter>,
+    private val numMonths: Int
+) : Metric<MetricResult.LatestSum> {
+
+    private var results: SortedSet<Instrument> = TreeSet(Comparator.reverseOrder())
```

Comment:
Думаю, тут лучше подойдет PriorityQueue c фиксированным размером. Из коробки такой нет, но можно можно просто удалять наименее приоритетный элемент из очереди, когда размер очереди будет превышен:
```suggestion
    private var results: PriorityQueue<Instrument> = PriorityQueue(Comparator.naturalOrder())

    override fun processInstrument(instrument: Instrument) {
        if (filters.all { it.check(instrument) }) {
            results.add(instrument)
        }
        if (results.size > numMonths) {
            results.poll()
        }
    }
```